src/states/adult/mod.rs

```diff
@@ -321,118 +322,52 @@ impl Adult {
         Ok(Transition::Stay)
     }
 
-    // Send signed_msg to our elders so they can route it properly.
-    fn send_signed_message_to_elders(
-        &mut self,
-        msg: HopMessageWithBytes,
-    ) -> Result<(), RoutingError> {
-        trace!(
-            "{}: Forwarding message {:?} via elder targets {:?}",
-            self,
-            msg.full_message_crypto_hash(),
-            self.chain.our_elders().format(", ")
-        );
-
-        let routing_msg_filter = &mut self.routing_msg_filter;
-        let targets: Vec<_> = self
-            .chain
-            .our_elders()
-            .filter(|p2p_node| {
-                routing_msg_filter
-                    .filter_outgoing(&msg, p2p_node.public_id())
-                    .is_new()
-            })
-            .map(|node| node.connection_info().clone())
-            .collect();
-
-        let cheap_bytes_clone = msg.full_message_bytes().clone();
-        self.send_message_to_targets(&targets, targets.len(), cheap_bytes_clone);
-
-        // we've seen this message - don't handle it again if someone else sends it to us
-        let _ = self.routing_msg_filter.filter_incoming(&msg);
-
-        Ok(())
-    }
-
     /// Handles a signature of a `SignedMessage`, and if we have enough to verify the signed
     /// message, handles it.
     fn handle_message_signature(
         &mut self,
         msg: AccumulatingMessage,
-        pub_id: PublicId,
+        src: PublicId,
+        outbox: &mut dyn EventBox,
     ) -> Result<Transition, RoutingError> {
-        if !self.chain.is_peer_elder(&pub_id) {
+        if !self.chain.is_peer_elder(&src) {
             debug!(
                 "{} - Received message signature from not known elder (still use it) {}, {:?}",
-                self, pub_id, msg
+                self, src, msg
             );
         }
 
-        if let Some(signed_msg) = self.sig_accumulator.add_proof(msg) {
-            let signed_msg = HopMessageWithBytes::new(signed_msg, &self.log_ident())?;
-            self.handle_signed_message(signed_msg)
+        if let Some(msg) = self.sig_accumulator.add_proof(msg, &self.log_ident()) {
+            self.handle_accumulated_message(msg, outbox)
         } else {
             Ok(Transition::Stay)
         }
     }
 
-    // If the message is for us, verify it then, handle the enclosed routing message and swarm it
-    // to the rest of our section when destination is targeting multiple; if not, forward it.
-    fn handle_signed_message(
+    fn handle_accumulated_message(
         &mut self,
-        msg: HopMessageWithBytes,
-    ) -> Result<Transition, RoutingError> {
-        if !self.routing_msg_filter.filter_incoming(&msg).is_new() {
+        msg: MessageWithBytes,
+        outbox: &mut dyn EventBox,
+    ) -> Result<Transition> {
+        if !self.filter_incoming_message(&msg) {
```

Comment:
did you meant to remove the duplication and call the base function? i.e https://github.com/maidsafe/routing/pull/2030#discussion_r372332158